ts/src/client.ts

```diff
@@ -179,7 +181,20 @@ class GrpcClient<TRequest extends ProtobufMessage, TResponse extends ProtobufMes
     }
 
     const grpcMessage = this.responseTrailers.get("grpc-message");
-    this.rawOnEnd(grpcStatus, grpcMessage[0], this.responseTrailers);
+    const statusMessage = this.decodeGRPCStatus(grpcMessage[0]);
+    this.rawOnEnd(grpcStatus, statusMessage, this.responseTrailers);
+  }
+
+  decodeGRPCStatus(src: string | undefined): string {
+    if (src) {
+      try {
+        return decodeURIComponent(src)
+      } catch (err) {
+        return src
+      }
+    } else {
+      return ""
```

Comment:
Is there a reason to return empty string here instead of `null` or `undefined` (undefined was the previous behaviour)? 

Appreciate that this is likely just opinion, but this is a subtle API change and the reason this PR is currently failing tests.